powerstore/resource_snapshot.go

```diff
@@ -306,10 +344,57 @@ func (r resourceVolumeSnapshot) updateSnapshotState(plan, state *models.Snapshot
 	} else {
 		state.ExpirationTimestamp = types.StringValue(expTime[:len(expTime)-6] + "Z")
 	}
-	state.CreatorType = types.StringValue(response.ProtectionData.CreatorType)
-	state.PerformancePolicyID = types.StringValue(response.PerformancePolicyID)
 	state.VolumeID = types.StringValue(response.ProtectionData.ParentID)
+	state.PerformancePolicyID = types.StringValue(response.PerformancePolicyID)
 	if plan != nil {
 		state.VolumeName = plan.VolumeName
 	}
 }
+
+func (r resourceVolumeSnapshot) planToServer(plan models.Snapshot) *gopowerstore.VolumeModify {
+	name := plan.Name.ValueString()
+	description := plan.Description.ValueString()
+	performancePolicyID := plan.PerformancePolicyID.ValueString()
+	expirationTimeStamp := plan.ExpirationTimestamp.ValueString()
+
+	volSnapshotUpdate := &gopowerstore.VolumeModify{
+		Description:         description,
+		Name:                name,
+		PerformancePolicyID: performancePolicyID,
+		ExpirationTimestamp: expirationTimeStamp,
+	}
+	return volSnapshotUpdate
+}
+
+// ModifyPlan modify resource plan attribute value
+func (r *resourceVolumeSnapshot) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	if req.Plan.Raw.IsNull() {
+		return
+	}
+	var plan models.Snapshot
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// if volume name is present instead of ID
+	if plan.VolumeID.ValueString() == "" {
```

Comment:
Please check if it results in the error for the scenario: volume name is used to create a snapshot and volume needs to be created first on the fly.

Reply:
Updated code to address this comment